process_data.sh

```diff
@@ -0,0 +1,298 @@
+    if [ ! -d "${bids_folder}"/derivatives/labels/"${SUBJECT}"/anat/ ]; then
+        mkdir -p "${bids_folder}"/derivatives/labels/"${SUBJECT}"/anat/
+    fi
+
+    # Go to folder where data will be copied and processed
+    cd "$results_folder"
+    # Copy source images
+    # Note: we use '/./' in order to include the sub-folder 'ses-0X'
+    rsync -Ravzh "$bids_folder"/./"$participant_id"/"$session_id" .
+}
+
+# Inspiration: https://github.com/spinalcordtoolbox/sct_tutorial_data/blob/master/multi_subject/process_data.sh#L66-L89
+segment_if_does_not_exist() {
+  ###
+  #  This function checks if a manual spinal cord segmentation file already exists, then:
+  #    - If it does, copy it locally.
+  #    - If it doesn't, perform automatic spinal cord segmentation.
+  #  This allows you to add manual segmentations on a subject-by-subject basis without disrupting the pipeline.
+  ###
+  local file="${1}"
+  local contrast="${2}"
+  # Update global variable with segmentation file name
+  FILESEG="${file}"_label-SC_seg
+  FILESEGMANUAL="${bids_folder}"/derivatives/labels/"${SUBJECT}"/anat/"${FILESEG}".nii.gz
+  echo
+  echo "Looking for manual segmentation: ${FILESEGMANUAL}"
+  if [[ -e "${FILESEGMANUAL}" ]]; then
+    echo "Found! Using manual segmentation."
+    rsync -avzh "${FILESEGMANUAL}" "${FILESEG}".nii.gz
+    sct_qc -i "${file}".nii.gz -s "${FILESEG}".nii.gz -p sct_deepseg_sc -qc "${PATH_QC}" -qc-subject "${SUBJECT}"
+  else
+    echo "Not found. Proceeding with automatic segmentation."
+    # Segment spinal cord
+    # TODO: consider replacing sct_deepseg_sc by contrast-agnostic or SCIseg as they might work better in compression
+    sct_deepseg_sc -i "${file}".nii.gz -c "${contrast}" -o "${FILESEG}".nii.gz -qc "${PATH_QC}" -qc-subject "${SUBJECT}"
+  fi
+}
+
+process_t2w()
+{
+    local suffix=$1
+    # Go to anat folder where all structural data are located
+    cd anat
+
+    # Construct the file name based on the subject ID, e.g., sub-001_ses-01_T2w
+    file_t2="${participant_id}_${session_id}_${suffix}"
+
+    # Segment spinal cord (only if it does not exist)
+    # TODO: redirect sct_deepseg_sc output to LOG file to do not clutter the users terminal
+    segment_if_does_not_exist "$file_t2" t2
+    file_t2_seg="${FILESEG}"
+
+    # Open FSLeyes to visualize the segmentation
+    echo "Opening FSLeyes (close FSLeyes to continue)..."
+    fsleyes "$file_t2".nii.gz "${file_t2_seg}.nii.gz" -cm red -a 70.0
+    # Copy the visually verified segmentation (and potentially manually corrected SC seg) to the derivatives folder
+    rsync -avzh "${file_t2_seg}".nii.gz "${bids_folder}"/derivatives/labels/"${SUBJECT}"/anat/
+    # TODO: continue with the analysis
+}
+
+main_analysis()
+{
+    # Create the results folder and copy the images to it
+    create_results_folder_and_copy_images
+
+    # Define path to the folder where QC will be stored
+    PATH_QC="${results_folder}"/qc
+    SUBJECT="${participant_id}/${session_id}"
```

Comment:
Define subject before calling creating the derivatives folder (this caused an error bellow)

```suggestion
    # Define path to the folder where QC will be stored
    PATH_QC="${results_folder}"/qc
    SUBJECT="${participant_id}/${session_id}"
    # Create the results folder and copy the images to it
    create_results_folder_and_copy_images

```

Reply:
Yes! Fixed in https://github.com/sct-pipeline/balgrist-sci/pull/5/commits/1ab668700eb4e3d29120aafa00fa89dbcb2487a5